File: tools-factory/src/main/java/com/koala/factory/extra/kugou/KugouSingnatureGenerator.java
package com.koala.factory.extra.kugou;

import com.koala.service.utils.MD5Utils;

import java.util.Arrays;
import java.util.Map;

import static com.koala.factory.secretKey.KugouSecretKeyCollector.KUGOU_PID_VERSION_SECRET_KEY;

public class KugouSingnatureGenerator {

    public static String generateKugouKey(String hash, String appId, String mid, String userId) {
        return MD5Utils.md5(nullToEmpty(hash) + KUGOU_PID_VERSION_SECRET_KEY + nullToEmpty(appId) + nullToEmpty(mid) + nullToEmpty(userId));
    }

    public static String generateKugouSignatureV2(String secret, Map<String, String> params) {
        if (params == null) {
            return null;
        }
        StringBuilder result = new StringBuilder(secret);
        for (Map.Entry<String, String> entry : params.entrySet()) {
            result.append(entry.getKey()).append("=").append(nullToEmpty(entry.getValue()));
        }
        return MD5Utils.md5(result + secret);
    }

    public static String generateKugouSignatureV1(String secret, Map<String, String> map) {
        if (map == null) {
            return "";
        }
        StringBuilder sb = new StringBuilder();
        String[] keys = map.keySet().toArray(new String[0]);
        Arrays.sort(keys);
        for (String key : keys) {
            sb.append(key);
            sb.append("=");
            sb.append(map.get(key));
        }
        return MD5Utils.md5(secret + sb + secret);
    }

    public static String getGetRequestParams(Map<String, String> map) {
        if (map == null) {
            return "";
        }
        StringBuilder sb = new StringBuilder("?");
        for (String key : map.keySet()) {
            sb.append(key);
            sb.append("=");
            sb.append(map.get(key));
            sb.append("&");
        }
        sb.deleteCharAt(sb.length() - 1);
        return sb.toString();
    }

    private static String nullToEmpty(String value) {
        return value == null ? "" : value;
    }
}

File: tools-factory/src/main/java/com/koala/factory/extra/kugou/KugouSingnatureGenerator.kt
package com.koala.factory.extra.kugou

import com.koala.factory.SecretKey.KugouSecretKeyCollector.KUGOU_ITEM_SECRET_KEY
import com.koala.factory.SecretKey.KugouSecretKeyCollector.KUGOU_PID_VERSION_SECRET_KEY
import com.koala.service.utils.MD5Utils
import java.util.*

class KugouSingnatureGenerator {

    companion object {
        @JvmStatic
        fun generateKugouKey(hash: String?, appId: String?, mid: String?, userId: String?): String {
            return MD5Utils.md5("${hash ?: ""}$KUGOU_PID_VERSION_SECRET_KEY${appId ?: ""}${mid ?: ""}${userId ?: ""}")
        }

        @JvmStatic
        fun generateKugouSignatureV2(params: HashMap<String, String?>?): String? {
            var result = ""
            params?.values?.let {
                result = "$result$KUGOU_ITEM_SECRET_KEY"
                for ((key, value) in params) {
                    result = "$result$key=${value ?: ""}"
                }
                return MD5Utils.md5("$result$KUGOU_ITEM_SECRET_KEY")
            }
            return null
        }

        @JvmStatic
        fun generateKugouSignatureV1(map: MutableMap<String?, String?>?): String? {
            if (map == null) {
                return ""
            }
            val sb = StringBuilder()
            val keySet = map.keys
            val strArr = keySet.toTypedArray<String?>()
            Arrays.sort<String?>(
                strArr,
                Comparator { obj: String?, anotherString: String? -> obj!!.compareTo(anotherString!!) })
            for (str3 in strArr) {
                sb.append(str3)
                sb.append("=")
                sb.append(map[str3])
            }
            return MD5Utils.md5(KUGOU_ITEM_SECRET_KEY + sb + KUGOU_ITEM_SECRET_KEY)
        }
    }

}